### DISCUSSION TITLE
Striping PII / censoring report messages

### DISCUSSION THREAD

**Discussion by g00g1** [fail2ban/fail2ban#3288] [Q&A]
Hello,

I am really curious if it is possible to gracefully strip any PII (like hostname, own IP addresses, domain names etc.) from `<matches>`? I know that I can do some `sed 's/example.com/redacted/g'` thing at actionban before reporting IP on AbuseIPDB or sending report to the blocklist.de, but are there any elegant solutions to do that kind of stuff?

**Comment by sebres** [comment]
> are there any elegant solutions to do that kind of stuff?

I guess at the moment no one "solution" can be considered as elegant or even good enough, but...

Fail2ban filters are able to capture groups using new tags like `<F-USER>...</F-USER>`, so if one know what exactly should be blacken in the message, one could define such groups and replace them in actions using `<F-USER>` tag (but it could surely replace something relevant by confusion with some equal content). Also they become always overwritten by the values captured from last matched message, so it will be not quite correct if some values changing by several attempts.
Alternatively, one could specify only relevant clean part of the message with `<F-REPORT>...</F-REPORT>` (or constructing a tuple with additional `<F-TUPLE_REPORT>...</F-TUPLE_REPORT>` by multiple parts in the message), so the tag `<F-REPORT>` can be then used instead of `<matches>` in action. But it'd also contain value of last message only.

To do it properly we'd need to extend fail2ban processing with special handling marking the parts of messages to be blaken (e. g. with tags like `<F-BLACK>`), as well as add such tags for the `prefregex` and `failregex` in any filters expecting that.

Does one really need to black that? I mean I would either avoid sending of matches at all (to some public services) or send them in original (to non-public services expecting that).
